src/linkedList.js

```diff
@@ -0,0 +1,170 @@
+'use strict'
+
+function Node(data) {
+  this.data = data
+  this.next = null
+}
+
+export default class LinkedList {
+  constructor() {
+    this.length = 0
+    this.head = null
+  }
+
+  isEmpty() {
+    return this.length === 0
+  }
+
+  size() {
+    return this.length
+  }
+
+  clear() {
+    this.length = 0
+    this.head = null
+  }
+
+  insert(data) {
+    var node = new Node(data)
+    var currentNode = this.head
+    if (!currentNode) {
+      this.head = node
+      this.length += 1
+      return node
+    }
+    while (currentNode.next) {
+      currentNode = currentNode.next
+    }
+    currentNode.next = node
+    this.length += 1
+    return node
+  }
+
+  getHeadNode(data) {
+    return this.head
+  }
+
+  contains(data) {
+    var node = new Node(data)
```

Comment:
`node` is unused.  See refactoring for the double linked list, above, for a more concise approach to `contains`.